x/pairing/types/geolocation.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	planstypes "github.com/lavanet/lava/x/plans/types"
+)
+
+// for convenience (calculate once only)
+var (
+	allGeoEnumRegionsList []int32
+	allGeoEnumRegions     int32
+)
+
+// initialize convenience vars at start-up
+func init() {
+	for _, geoloc := range planstypes.Geolocation_value {
+		if geoloc != int32(planstypes.Geolocation_GLS) && geoloc != int32(planstypes.Geolocation_GL) {
+			allGeoEnumRegionsList = append(allGeoEnumRegionsList, geoloc)
+			allGeoEnumRegions |= geoloc
+		}
+	}
+}
+
+// IsValidGeoEnum tests the validity of a given geolocation
+func IsValidGeoEnum(geoloc int32) bool {
+	return geoloc != int32(planstypes.Geolocation_GLS) && (geoloc & ^allGeoEnumRegions) == 0
```

Comment:
why not just try to fetch from the map that was generated in the pb.go?
planstypes.Geolocation_name[geoloc]

Reply:
done. see https://github.com/lavanet/lava/pull/577/commits/99b8a4547637888ed9304f1e623eeaf1637d8cc9

Reply:
Reverted